backend/de.metas.manufacturing.rest-api/src/main/java/de/metas/manufacturing/job/service/ManufacturingJobService.java

```diff
@@ -509,29 +515,33 @@ private ManufacturingJob issueRawMaterialsInTrx(final @NonNull ManufacturingJob
 	}
 
 	public ManufacturingJob receiveGoods(
+			@NonNull final JsonManufacturingOrderEvent.ReceiveFrom receiveFrom,
 			@NonNull final ManufacturingJob job,
-			@NonNull final FinishedGoodsReceiveLineId lineId,
-			@NonNull final SelectedReceivingTarget receivingTarget,
-			@NonNull final BigDecimal qtyToReceiveBD,
 			@NonNull final ZonedDateTime date)
 	{
-		final ManufacturingJob changedJob = job.withChangedReceiveLine(lineId, line -> {
-			final ReceivingTarget newReceivingTarget = trxManager.callInThreadInheritedTrx(() -> ReceiveGoodsCommand.builder()
-					.handlingUnitsBL(handlingUnitsBL)
-					.ppOrderBL(ppOrderBL)
-					.ppOrderBOMBL(ppOrderBOMBL)
-					.loadingAndSavingSupportServices(loadingAndSavingSupportServices)
-					//
-					.ppOrderId(job.getPpOrderId())
-					.coProductBOMLineId(line.getCoProductBOMLineId())
-					.receivingTarget(receivingTarget)
-					.qtyToReceiveBD(qtyToReceiveBD)
-					.date(date)
-					//
-					.build().execute());
-
-			return line.withReceivingTarget(newReceivingTarget);
-		});
+		final SelectedReceivingTarget receivingTarget = SelectedReceivingTarget.builder()
+				.luReceivingTarget(receiveFrom.getAggregateToLU())
+				.tuReceivingTarget(receiveFrom.getAggregateToTU())
+				.build();
+
+		final ManufacturingJob changedJob = job.withChangedReceiveLine(receiveFrom.getFinishedGoodsReceiveLineId(), line -> trxManager.callInThreadInheritedTrx(() -> ReceiveGoodsCommand.builder()
+				.handlingUnitsBL(handlingUnitsBL)
+				.ppOrderBL(ppOrderBL)
+				.ppOrderBOMBL(ppOrderBOMBL)
+				.uomConversionBL(uomConversionBL)
+				.huPIItemProductBL(hupiItemProductBL)
+				.loadingAndSavingSupportServices(loadingAndSavingSupportServices)
+				//
+				.ppOrderId(job.getPpOrderId())
+				.receiveLine(line)
```

Comment:
IMHO there is no need to provide `receiveLine` because that will couple the ReceiveGoodsCommand even more.
better,
* introduce ReceiveGoodsRequest (as I was suggested somewhere else)
* introduce ReceiveGoodsResponse which will be returned by ReceiveGoodsCommand.execute and will contain receivingTarget and qtyReceived.